blockchain/internal/spendpruner/pruner.go

```diff
@@ -0,0 +1,351 @@
+// Copyright (c) 2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package spendpruner
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/database/v3"
+)
+
+// spendPrunerEventType represents a spend pruner event message.
+type spendPrunerEventType int
+
+const (
+	// SPBlockConnected indicates a new block was connected to the chain.
+	SPBlockConnected spendPrunerEventType = iota
```

Comment:
It's not a big deal since it's an internal package, but these should not be exported since they are only used internally.

Reply:
noted, updating.